device/topology/topology_discovery.cpp

```diff
@@ -104,13 +104,26 @@ void TopologyDiscovery::discover_remote_chips() {
             chip->get_soc_descriptor().get_cores(CoreType::ETH, umd_use_noc1 ? CoordSystem::NOC1 : CoordSystem::NOC0);
         TTDevice* tt_device = chip->get_tt_device();
 
+        std::vector<uint32_t> intermesh_eth_links;
+        if (eth_cores.size() > 0) {
+            intermesh_eth_links = extract_intermesh_eth_links(chip, eth_cores.front());
+        }
+
         uint32_t channel = 0;
         for (const CoreCoord& eth_core : eth_cores) {
             uint32_t port_status = read_port_status(chip, eth_core);
 
             if (is_eth_unknown(chip, eth_core) || is_eth_unconnected(chip, eth_core)) {
-                channel++;
-                continue;
+                if (std::find(intermesh_eth_links.begin(), intermesh_eth_links.end(), channel) !=
+                    intermesh_eth_links.end()) {
+                    if (!is_intermesh_eth_link_trained(chip, eth_core)) {
+                        channel++;
+                        continue;
+                    }
+                } else {
+                    channel++;
+                    continue;
+                }
```

Comment:
I find it more readable and easier to follow the logic if not nested, and because of continue you don't need an else branch

```suggestion
                if (std::find(intermesh_eth_links.begin(), intermesh_eth_links.end(), channel) ==
                    intermesh_eth_links.end()) {
                    channel++;
                    continue;
                }
                if (!is_intermesh_eth_link_trained(chip, eth_core)) {
                    channel++;
                    continue;
                }
```